server/apis/v1/handler.go

```diff
@@ -969,11 +952,20 @@ func validatePipelineSpec(h *handler, oldPipeline *dfv1.Pipeline, newPipeline *d
 }
 
 // validateISBSVCSpec is used to validate the ISB service spec
-func validateISBSVCSpec(h *handler, prevSpec *dfv1.InterStepBufferService,
+func validateISBSVCSpec(prevSpec *dfv1.InterStepBufferService,
 	newSpec *dfv1.InterStepBufferService, validType string) error {
-	ns := newSpec.Namespace
-	isbClient := h.numaflowClient.InterStepBufferServices(ns)
-	valid := validator.NewISBServiceValidator(h.kubeClient, isbClient, prevSpec, newSpec)
+	// UI-specific validations: updating namespace and name from UI is not allowed
+	if validType == ValidTypeUpdate && prevSpec != nil {
+		if prevSpec.Namespace != newSpec.Namespace {
```

Comment:
What happens if `namespace` is not specified in the new spec?

Reply:
Thanks for catching this edge case. If a user doesn't specify the namespace, it will be an empty string. K8S treats empty namespace as the same as "default". I updated the PR to reflect this.

Reply:
This is not right. 

When you create an object in a namespace, if namespace filed is missing, it should use the specified namespace.

Reply:
Yes exactly. When creating, I implemented it to use the specified namespace.

Reply:
To summarize, if a user creates a brand-new spec and the namespace field is missing, we use the namespace from the URL parameter. If a user updates an existing spec, if the namespace field is missing, it means the user purposely removed the namespace field, in this case, we assume the user wants to use the "default" namespace and validate it against the previous namespace.

Let me know if this logic needs to be revised. @whynowy

Reply:
I think the logic for updating should be:

If namespace is missing, we always use the namespace in the URL - which means we do not need to validate it. This is the same behavior as we using kubectl to update something.